cmd/clusterctl/client/cluster/client.go

```diff
@@ -228,6 +231,9 @@ type Proxy interface {
 	// ValidateKubernetesVersion returns an error if management cluster version less than minimumKubernetesVersion
 	ValidateKubernetesVersion() error
 
+	// ValidateKubernetesMaxVersion returns an error if management cluster version is greater than or equal to firstUnsupportedKubernetesVersion
```

Comment:
I'm not sure if it's the best idea to add an additional func. The problem is that `ValidateKubernetesVersion()` is currently only used to check if CRDs can be deployed and it wouldn't make sense to verify serverVersion < v1.22.0 in this case.